### PR TITLE
Problem: Failed to upgrade `prost-0.11` and `tonic-0.8` (fix #543 and #544)

### LINKED ISSUE #543: Problem: upgrade to tonic 0.8 fails
https://github.com/crypto-com/defi-wallet-core-rs/runs/7627450319?check_suite_focus=true

### LINKED ISSUE #544: Problem: upgrade to prost 0.11 fails
https://github.com/crypto-com/defi-wallet-core-rs/runs/7627446860?check_suite_focus=true

### PR DESCRIPTION
**Pull request by stevenatcrypto**
Close #543 
Close #544 

# PR Checklist:

- [x] Have you read the [CONTRIBUTING.md](https://github.com/crypto-com/defi-wallet-core-rs/blob/main/CONTRIBUTING.md)?
- [x] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [x] Have you rebased your work on top of the latest main? 
- [x] Have you checked your code compiles? (`cargo build`)
- [x] Have you included tests for any non-trivial functionality?
- [x] Have you checked your code passes the unit tests? (`cargo test`)
- [x] Have you checked your code formatting is correct? (`cargo fmt -- --check --color=auto`)
- [x] Have you checked your basic code style is fine? (`cargo clippy`)
- [x] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`cargo audit`)
- [x] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [x] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-com/defi-wallet-core-rs/blob/main/CHANGELOG.md)?
- [x] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-com/defi-wallet-core-rs/blob/main/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-com/defi-wallet-core-rs/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/src/transaction/cosmos_sdk/parser/structs/cosmos_raw_msg.rs [hunk 1]
```diff
@@ -410,7 +409,7 @@ impl CosmosRawNormalMsg {
     }
 }
 
-#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
```
Comment (tomtau) [inline common/src/transaction/cosmos_sdk/parser/structs/cosmos_raw_msg.rs:412]: if `Eq` is now derived in common, the clippy ignore can be removed?

**Review by tomtau — APPROVED** [review]
seems ok, but the manual feature guards in generated protobuf may need to be kept

File: common/src/transaction/cosmos_sdk/parser/structs/cosmos_raw_msg.rs (hunk 1 above)
Comment (stevenatcrypto) [reply common/src/transaction/cosmos_sdk/parser/structs/cosmos_raw_msg.rs:412]: Deleted `#![allow(clippy::derive_partial_eq_without_eq)]` in `common/src/lib.rs`.

File: proto/src/prost/terra.market.v1beta1.rs [hunk 2]
```diff
@@ -40,13 +40,12 @@ pub struct MsgSwapSendResponse {
...
     #![allow(unused_variables, dead_code, missing_docs, clippy::let_unit_value)]
     use tonic::codegen::*;
+    use tonic::codegen::http::Uri;
     /// Msg defines the market Msg service.
     #[derive(Debug, Clone)]
     pub struct MsgClient<T> {
         inner: tonic::client::Grpc<T>,
     }
-    // Workaround, add feature manually, it could be fixed after https://github.com/hyperium/tonic/issues/491
-    #[cfg(feature = "transport")]
```
[the remark this answers is not in the document]
Comment (stevenatcrypto) [reply proto/src/prost/terra.market.v1beta1.rs]: (Revert to) Added this feature-guard to all terra proto files.

**Pull request merged**